src/SphereXConfiguration.sol

```diff
@@ -0,0 +1,201 @@
+     */
+    function sphereXEngine() public view returns (address) {
+        return _getAddress(SPHEREX_ENGINE_STORAGE_SLOT);
+    }
+
+    /**
+     * Setting the address of the next admin. this address will have to accept the role to become the new admin.
+     * @dev Could not use OZ Ownable2Step because the client's contract might use it.
+     */
+    function transferSphereXAdminRole(address newAdmin) public virtual onlySphereXAdmin {
+        _setAddress(SPHEREX_PENDING_ADMIN_STORAGE_SLOT, newAdmin);
+        emit SpherexAdminTransferStarted(sphereXAdmin(), newAdmin);
+    }
+
+    /**
+     * Accepting the admin role and completing the transfer.
+     * @dev Could not use OZ Ownable2Step because the client's contract might use it.
+     */
+    function acceptSphereXAdminRole() public virtual {
+        require(pendingSphereXAdmin() == msg.sender, "SphereX error: not the pending account");
+        address oldAdmin = sphereXAdmin();
+        _setAddress(SPHEREX_ADMIN_STORAGE_SLOT, msg.sender);
+        _setAddress(SPHEREX_PENDING_ADMIN_STORAGE_SLOT, address(0));
+        emit SpherexAdminTransferCompleted(oldAdmin, msg.sender);
+    }
+
+    /**
+     *
+     * @param newSphereXOperator new address of the new operator account
+     */
+    function changeSphereXOperator(address newSphereXOperator) external onlySphereXAdmin {
+        address oldSphereXOperator = _getAddress(SPHEREX_OPERATOR_STORAGE_SLOT);
+        _setAddress(SPHEREX_OPERATOR_STORAGE_SLOT, newSphereXOperator);
+        emit ChangedSpherexOperator(oldSphereXOperator, newSphereXOperator);
+    }
+
+    /**
+     * Checks the given address implements ISphereXEngine or is address(0)
+     * @param newSphereXEngine new address of the spherex engine
+     */
+    function _checkSphereXEngine(address newSphereXEngine) private view {
+        require(
+            newSphereXEngine == address(0)
+                || ISphereXEngine(newSphereXEngine).supportsInterface(type(ISphereXEngine).interfaceId),
+            "SphereX error: not a SphereXEngine"
+        );
+    }
+
+    /**
+     *
+     * @param newSphereXEngine new address of the spherex engine
+     * @dev this is also used to actually enable the defense
+     * (because as long is this address is 0, the protection is disabled).
+     */
+    function changeSphereXEngine(address newSphereXEngine) external spherexOnlyOperator {
+        _checkSphereXEngine(newSphereXEngine);
+        address oldEngine = _getAddress(SPHEREX_ENGINE_STORAGE_SLOT);
+        _setAddress(SPHEREX_ENGINE_STORAGE_SLOT, newSphereXEngine);
+        emit ChangedSpherexEngineAddress(oldEngine, newSphereXEngine);
+    }
+    // ============ Engine interaction ============
+
+    function _addAllowedSenderOnChain(address newSender) internal {
+        ISphereXEngine engine = _sphereXEngine();
+        if (address(engine) != address(0)) {
+            engine.addAllowedSenderOnChain(newSender);
+        }
+        emit NewAllowedSenderOnchain(newSender);
```

Comment:
this is weird that it is outside of the if

Reply:
You are correct. Fixed. (also, I think it's not my code... I think I copied from the protectedBase)